Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -168,6 +220,175 @@ namespace PhysX
         return AZ::Edit::PropertyRefreshLevels::None;
     }
 
+    AZ::u32 EditorHeightfieldColliderComponent::SaveHeightfieldAssetToDisk()
+    {
+        auto assetType = AZ::AzTypeInfo<PhysX::Pipeline::HeightFieldAsset>::Uuid();
+        auto assetHandler = const_cast<AZ::Data::AssetHandler*>(AZ::Data::AssetManager::Instance().GetHandler(assetType));
+        if (assetHandler)
+        {
+            char projectPath[AZ_MAX_PATH_LEN];
+            AZ::IO::FileIOBase::GetInstance()->ResolvePath("@projectroot@", projectPath, AZ_MAX_PATH_LEN);
+
+            AZStd::string heightfieldFullPath;
+            AzFramework::StringFunc::Path::ConstructFull(projectPath, m_bakedHeightfieldRelativePath.c_str(), heightfieldFullPath, true);
+
+            AZ::IO::FileIOStream fileStream(heightfieldFullPath.c_str(), AZ::IO::OpenMode::ModeWrite);
+            if (fileStream.IsOpen())
+            {
+                if (!assetHandler->SaveAssetData(m_bakedHeightfieldAsset, &fileStream))
+                {
+                    AZ_Error("PhysX", false, "Unable to save heightfield asset %s", heightfieldFullPath.c_str());
+                }
+            }
+        }
+        return 0;
+    }
+
+    void EditorHeightfieldColliderComponent::StartHeightfieldBakingJob()
+    {
+        FinishHeightfieldBakingJob();
+
+        m_bakingCompletion.Reset(true /*isClearDependent*/);
+        m_heightfieldAssetBakingJob.Reset(true);
+
+        m_heightfieldAssetBakingJob.SetDependent(&m_bakingCompletion);
+        m_heightfieldAssetBakingJob.Start();
+    }
+
+    void EditorHeightfieldColliderComponent::FinishHeightfieldBakingJob()
+    {
+        m_bakingCompletion.StartAndWaitForCompletion();
+    }
+
+    bool EditorHeightfieldColliderComponent::CheckHeightfieldPathExists()
+    {
+        if (!m_bakedHeightfieldRelativePath.empty())
+        {
+            char projectPath[AZ_MAX_PATH_LEN];
+            AZ::IO::FileIOBase::GetInstance()->ResolvePath("@projectroot@", projectPath, AZ_MAX_PATH_LEN);
+
+            // retrieve the source heightfield path from the configuration
+            // we need to make sure to use the same source heightfield for each baking
+
+            // test to see if the heightfield file is actually there, if it was removed we need to
+            // generate a new filename, otherwise it will cause an error in the asset system
+            AZStd::string fullPath;
+            AzFramework::StringFunc::Path::Join(projectPath,
+                m_bakedHeightfieldRelativePath.c_str(), fullPath,
+                /*bCaseInsensitive*/true, /*bNormalize*/true);
+
+            if (!AZ::IO::FileIOBase::GetInstance()->Exists(fullPath.c_str()))
```

Comment:
I've been digging into the `FileIOBase` a lot lately, this would have worked :)
It's submitted now, but FYI if in the future you have to do further changes.